src/jit.c

```diff
@@ -0,0 +1,1550 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+/* This JIT implementation has undergone extensive modifications, heavily
+ * relying on the ubpf_jit_[x86_64|arm64].[ch] from ubpf. The original
+ * ubpf_jit_[x86_64|arm64].[ch] file served as the foundation and source of
+ * inspiration for adapting and tailoring it specifically for this JIT
+ * implementation. Therefore, credit and sincere thanks are extended to ubpf for
+ * their invaluable work.
+ *
+ * Reference:
+ *   https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_x86_64.c
+ *   https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_arm64.c
+ */
+
+#if !defined(__x86_64__) && !defined(__aarch64__)
+#error "This implementation is dedicated to x64 and arm64."
+#endif
+
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#if HAVE_MMAP
+#include <sys/mman.h>
+#include <unistd.h>
+#else
+/* fallback to standard I/O text stream */
```

Comment:
Don't do that. There should be no fallback because we support *BSD (macOS included) and Linux.